hathor/merged_mining/coordinator.py

```diff
@@ -978,11 +999,32 @@ class MergedJob(NamedTuple):
     payback_script_bitcoin: Optional[bytes]
     clean: bool
 
+    def build_sample_block_proposal(self) -> bytes:
+        return bytes(self._new_single_miner_job(
+            '',
+            b'\0' * MergedMiningCoordinator.XNONCE1_SIZE,
+            MergedMiningStratumProtocol.DEFAULT_XNONCE2_SIZE,
+            b'',
+            b'',
+        ).dummy_bitcoin_block())
+
     def new_single_miner_job(self, protocol: MergedMiningStratumProtocol) -> SingleMinerJob:
         """ Generate a partial job for a single miner, based on this job.
         """
-        # payback_address_bitcoin = protocol.coordinator.payback_address_bitcoin
-        xnonce_size = len(protocol.xnonce1) + protocol.xnonce2_size
+        assert protocol.payback_script_hathor is not None
+        payback_script_bitcoin = self.payback_script_bitcoin or protocol.payback_script_bitcoin
+        assert payback_script_bitcoin is not None
+        return self._new_single_miner_job(
+            protocol.next_job_id(),
+            protocol.xnonce1,
+            protocol.xnonce2_size,
+            protocol.payback_script_hathor,
+            payback_script_bitcoin,
+        )
+
+    def _new_single_miner_job(self, job_id: str, xnonce1: bytes, xnonce2_size: int, payback_script_hathor: bytes,
+                              payback_script_bitcoin: bytes) -> SingleMinerJob:
```

Comment:
Missing docstring. Why did you split the code in a separate method if it is called in only one place?

Reply:
No, it's called right above on build_sample_block_proposal too.

Reply:
Added docstring.

Reply:
Ok. Thanks!